### PR TITLE
feat(terraform): Adding yaml based build time policies for corresponding PC runtime policies

### PR DESCRIPTION
**Pull request by praveen-panw**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

This PR has 4 YAML-based checks with detailed description below

### 1. Checkov Title - Ensure Azure AD authentication is enabled for Azure SQL (MSSQL)

<ins>PC Policy ID</ins> - 4d2615bb-091e-48fd-87b7-77a277d7d2fd

<ins>PC Policy Title</ins> - Azure SQL server not configured with Active Directory admin authentication

<ins>Compliance standards </ins> - 

`PCI DSS v3.2.1` `NIST 800-53 Rev4` `CIS v1.4.0 (Azure)` `CIS Controls v7.1` `ISO 27001:2013` `Copy of CSA CCM v3.0.1` `CSA CCM v3.0.1` `CIS v1.1 (Azure)` `SOC 2` `HITRUST CSF v9.3` `PIPEDA` `NIST CSF` `NIST SP 800-172` `CCPA 2018` `Azure Security Benchmark (v3)` `Copy of APRA (CPS 234) Information Security` `mkurter-CIS Controls v8-testing` `CIS v1.3.1 (Azure)` `Azure Security Benchmark (v2)` `CIS v1.5.0 (Azure) - Level 1` `MLPS 2.0 (Level 2)` `CIS Controls v8` `Australian Energy Sector Cyber Security Framework (AESCSF)` `ACSC Information Security Manual (ISM)` `Secure Controls Framework (SCF) - 2022.2.1` `Copy of ACSC Information Security Manual (ISM)` `Sarbanes Oxley Act (SOX)` `CyberSecurity Law of the People's Republic of China` `CIS v1.2.0 (Azure)` `APRA (CPS 234) Information Security` `Cybersecurity Maturity Model Certification (CMMC) v.1.02` `CIS v1.3.0 (Azure)` `Brazilian Data Protection Law (LGPD)` `NIST SP 800-171 Revision 2`

<ins>Remediation:</ins>

1. Log in to Azure Portal
2. Navigate SQL servers dashboard
3. Select reported each SQL server
4. Click on Azure Active Directory (under 'Settings')
5. Click on 'Set admin'
6. Select an Azure Active Directory from available options
7. Click on Select
8. Click on Save

---

### 2. Checkov Title - Ensure Azure PostgreSQL Flexible server is not configured with overly permissive network access

<ins>PC Policy ID</ins> - cdb612cb-5629-4e0b-a028-8ce69d01104c

<ins>PC Policy Title</ins> - Azure PostgreSQL database flexible server configured with overly permissive network access

<ins>Compliance standards</ins> - 

`MLPS 2.0 (Level 2)` `Secure Controls Framework (SCF) - 2022.2.1` `Sarbanes Oxley Act (SOX)`

<ins>Remediation</ins>:

If PostgreSQL database flexible server was intended to accessed by authorized public IPs. Restrict IP addresses to known list and make sure IP range '+ Add 0.0.0.0 - 255.255.255.255' is not in Firewall rules. 
To add or to remove IPs refer below URL:
https://docs.microsoft.com/en-gb/azure/postgresql/flexible-server/how-to-manage-firewall-portal#manage-existing-firewall-rules-through-the-azure-portal

To create new PostgreSQL database flexible server with Private access (VNet Integration), refer below URL:
https://docs.microsoft.com/en-gb/azure/postgresql/flexible-server/quickstart-create-server-portal

Note: Once PostgreSQL database flexible server is created; You can't change the connectivity method after. For example, if you select Public access (allowed IP addresses) when you create the server, you can't change to Private access (VNet Integration) after the server is created.

---

### 3. Checkov Title - Ensure MySQL DB instance backup binary logs is configured

<ins>PC Policy ID</ins> - 25f41a54-717e-410c-a2ae-fb293dc2ed6d

<ins>PC Policy Title</ins> - Ensure MySQL DB instance has point-in-time recovery backup configured

<ins>Compliance standards</ins> - 

`HIPAA` `PCI DSS v3.2.1` `NIST 800-53 Rev4` `ISO 27001:2013` `SOC 2` `CIS Controls v7.1` `CSA CCM v.4.0.1` `ISO/IEC 27002:2013` `Copy of APRA (CPS 234) Information Security` `mkurter-CIS Controls v8-testing` `Fedramp (Moderate)` `New Zealand Information Security Manual (NZISM v3.4)` `CIS Controls v8` `Australian Cyber Security Centre (ACSC) Essential Eight` `Australian Energy Sector Cyber Security Framework (AESCSF)` `ACSC Information Security Manual (ISM)` `Secure Controls Framework (SCF) - 2022.2.1` `Copy of ACSC Information Security Manual (ISM)` `Sarbanes Oxley Act (SOX)` `APRA (CPS 234) Information Security` `Cybersecurity Maturity Model Certification (CMMC) v.1.02` `Brazilian Data Protection Law (LGPD)` `NIST SP 800-171 Revision 2` `NIST SP 800-172` `NIST CSF` `ISO/IEC 27017:2015`

<ins>Remediation:</ins>

Follow the below mentioned URL to enable point-in-time recovery backup (Binary logs) for the reported MySQL instance:

https://cloud.google.com/sql/docs/mysql/backup-recovery/pitr

---

### 4. Checkov Title - Ensure GCP Kubernetes engine clusters have 'alpha cluster' feature disabled

<ins>PC Policy ID</ins> - d49d0b8f-790f-412c-b59f-9ccdd0bca8f8

<ins>PC Policy Title</ins> - GCP Kubernetes Engine Clusters have Alpha cluster feature enabled

<ins>Compliance standards</ins> - 

`CIS Controls v7.1` `PIPEDA` `CCPA 2018` `CIS v1.1.0 (GKE)` `CSA CCM v.4.0.1` `ISO/IEC 27002:2013` `Copy of APRA (CPS 234) Information Security` `mkurter-CIS Controls v8-testing` `New Zealand Information Security Manual (NZISM v3.4)` `CIS Controls v8` `FFIEC` `PCI DSS v4.0` `CIS v1.2.0 (GKE)` `ACSC Information Security Manual (ISM)` `Secure Controls Framework (SCF) - 2022.2.1` `Copy of ACSC Information Security Manual (ISM)` `Sarbanes Oxley Act (SOX)` `APRA (CPS 234) Information Security` `Cybersecurity Maturity Model Certification (CMMC) v.1.02` `CIS v1.3.0 (GKE) - Level 1` `Brazilian Data Protection Law (LGPD)` `NIST SP 800-171 Revision 2` `PCI DSS v3.2.1` `NIST SP 800-172` `HITRUST v.9.4.2` `NIST CSF` `ISO/IEC 27017:2015` `ISO/IEC 27018:2019`

<ins>Remediation</ins>:

GCP Kubernetes Engine Clusters alpha feature cannot be disabled once it is created. So to resolve this alert, create a new cluster with the alpha feature disabled, then migrate all required cluster data from the reported cluster to this newly created cluster and delete reported Kubernetes engine cluster.

To create new Kubernetes engine cluster with the alpha feature disabled, perform the following: 
1. Login to GCP Portal
2. Go to Kubernetes Engine (Left Panel)
3. Select Kubernetes clusters 
4. Click on CREATE CLUSTER button
5. Set new cluster parameters as per your requirement and make sure 'Enable Kubernetes alpha features in this cluster' is unchecked.
6. Click on Save

To delete reported Kubernetes engine cluster, perform the following:
1. Login to GCP Portal
2. Go to Kubernetes Engine (Left Panel)
3. Select Kubernetes clusters 
4. Click on reported Kubernetes cluster
5. Click on the DELETE button
6. On 'Delete a cluster' popup dialog, Click on DELETE to confirm the deletion of the cluster.

---

### CODE REVIEW

File: checkov/terraform/checks/graph_checks/gcp/GCPdisableAlphaClusterFeatureInKubernetesEngineClusters.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+metadata:
+  id: "CKV2_GCP_19"
+  name: "Ensure GCP Kubernetes engine clusters have 'alpha cluster' feature disabled"
+  category: "KUBERNETES"
+
+definition:
+  cond_type: attribute
+  resource_types: "google_container_cluster"
+  attribute: "enable_kubernetes_alpha"
+  operator: "equals_ignore_case"
```
Comment (gruebel) [inline checkov/terraform/checks/graph_checks/gcp/GCPdisableAlphaClusterFeatureInKubernetesEngineClusters.yaml:10]:
```suggestion
  operator: "not_equals_ignore_case"
```
it should not `true` right?

**Review by gruebel — COMMENTED** [review]
nice work, just one comment 🙂

File: checkov/terraform/checks/graph_checks/gcp/GCPdisableAlphaClusterFeatureInKubernetesEngineClusters.yaml (hunk 1 above)
Comment (praveen-panw) [reply checkov/terraform/checks/graph_checks/gcp/GCPdisableAlphaClusterFeatureInKubernetesEngineClusters.yaml:10]: Oops, that's right. Making the changes now.

**Review by gruebel — APPROVED** [review]
great work 🥇

**Review by tronxd — COMMENTED** [review]
The tests won't run unless you add them to `test_yaml_policies.py` as a new test case

**Review by tronxd — APPROVED** [review]

**Pull request merged**